src/webapp/components/app-bar/GlassAppBar.tsx

```diff
@@ -37,11 +39,23 @@ const useStyles = makeStyles(theme => ({
 export const GlassAppBar: React.FC = () => {
     const classes = useStyles();
 
-    const [country, setCountry] = React.useState(1);
+    const { currentUser } = useAppContext();
+    const { currentOrgUnitAccess, changeCurrentOrgUnitAccess } = useCurrentOrgUnitContext();
     const [action, setAction] = React.useState(1);
 
-    const changeCountry = (event: React.ChangeEvent<{ value: unknown }>) => {
-        setCountry(event.target.value as number);
+    const [orgUnit, setOrgUnit] = React.useState(currentOrgUnitAccess.orgUnitName);
+
+    useEffect(() => {
+        if (orgUnit !== currentOrgUnitAccess.orgUnitName) {
+            //if orgUnit has been changed manually in url
+            setOrgUnit(currentOrgUnitAccess.orgUnitName);
+        }
+    }, [orgUnit, setOrgUnit, currentOrgUnitAccess.orgUnitName]);
+
+    const changeOrgUnit = (e: React.ChangeEvent<{ name?: string | undefined; value: unknown }>) => {
+        if (e.target?.value) setOrgUnit(e.target?.value as string);
+        const orgUnitId = (e.currentTarget as HTMLInputElement).getAttribute("data-key");
```

Comment:
The orthodox react approach does not use `target` or get data from the attributes. You pass whatever values you need with JS (typically, with arguments)

Reply:
We need the key using e.currentTarget.getAttribute("data-key"), however for some reason e.currentarget is not available unless typed. Maybe due to material-ui Select implementation - could not find a way to get the required data without passing the event as a whole.

Reply:
You are right about `e.target.value`, I see that's the way MUI select works. 

However, we should avoid data-key. Typically, the items in a selector have value and text. orgUnit.id -> value, orgUnit.name -> text. And you have a mapping to go from one to another when necessary. Here we are using only the name.

This is not a major problem, if we have time constraint we can refactor it at a later time.